corehq/form_processor/models.py

```diff
@@ -455,7 +458,7 @@ def is_deleted(self):
         return self.deleted
 
     def dynamic_case_properties(self):
-        return self.case_json
+        return OrderedDict(sorted(self.case_json.iteritems()))
```

Comment:
`CommCareCase` [does it](https://github.com/dimagi/commcare-hq/blob/fd0a33742e65a1c19be383925a91d08521eb7e5e/corehq/ex-submodules/casexml/apps/case/models.py#L498), and one of the tests was failing without it because the dynamic properties in a restore payload were in a different order. A better solution might be to change the test to be order agnostic?